script.js

```diff
@@ -1,4 +1,211 @@
+      return `Please enter a valid integer.`;
+    }
+
+    var diceCombined = arrangeDice(arrayActive, input);
+
+    // Create new player on scoreboard if needed, else simply add the
+    // value to the correct player
+    if (scoreBoardLock == 0) {
+      scoreBoard.push(diceCombined);
+    } else {
+      scoreBoard[currentPlayer - 1] += diceCombined;
+    }
+
+    // If there are only two players, and both players have played compare their scores and print a winner. Else, empty the field.
+    if (numberOfPlayers == 2 && lastPlayer != 0) {
+      checkWinner = determineWinner(
+        lastPlayer,
+        currentPlayer,
+        scoreBoard[lastPlayer - 1],
+        scoreBoard[currentPlayer - 1]
+      );
+    } else {
+      checkWinner = "";
+    }
+
+    currentLeaderBoard = leaderBoard(scoreBoard);
+    rollingPlayer = currentPlayer;
+
+    // If all players have played, reset the game and lock the scoreboard
+    // else, update for the next player
+    if (currentPlayer == numberOfPlayers) {
+      lastPlayer = 0;
+      currentPlayer = 1;
+      scoreBoardLock = 1;
+    } else {
+      lastPlayer = currentPlayer;
+      currentPlayer += 1;
+    }
+
+    // Update to loop game state
+    gameState = "roll";
+
+    return `The current score is ${scoreBoard}.<br>Player ${rollingPlayer} rolled ${diceCombined}.<br> ${checkWinner}
+    <br> ${currentLeaderBoard}`;
+  }
+};
+
+// Helper functions
+
+var inputValidator = function (input) {
+  // Return true if it's a number
+  if (!isNaN(input) && input != "") {
+    return true;
+  }
+
+  // Return false if it's not a number
+  return false;
+};
+
+// Return random number from 1 to 6
+var rollDice = function () {
+  // Generate a decimal from 0 to 5 inclusive
+  var randomDecimal = Math.random() * 6;
+  var randomInteger = Math.floor(randomDecimal);
+
+  // Convert to provide a result from 1 to 6
+  var randomResult = randomInteger + 1;
+
+  return randomResult;
+};
+
+// Return an array containing as many dice rolls as input
+var generateDiceArray = function (diceRolls) {
+  var generatedArray = [];
+  // Execute the roll dice function and push its result to an array based on the number of diceRolls
+  for (let i = 0; i < diceRolls; i += 1) {
+    generatedArray.push(rollDice());
+  }
+  return generatedArray;
+};
+
+// Rearrange dice and return integer
+var arrangeDice = function (diceArray, diceChoice) {
+  // Arrange dice automatically if autoPlay is on
+  if (autoPlay == 1) {
+    if (diceArray[0] > diceArray[1]) {
+      return diceArray[0] * 10 + diceArray[1];
+    }
+    return diceArray[1] * 10 + diceArray[0];
+  }
+
+  // Take the chosen number as the first number, then remove it from the array
+  var firstNumber = diceArray[diceChoice - 1];
+  diceArray.splice(diceChoice - 1, 1);
+
+  // Take the remaining number as the second number
+  var secondNumber = diceArray[0];
+  var diceValue = firstNumber * 10 + secondNumber;
+
+  return diceValue;
+};
```

Comment:
Great job in coding the logic for concatenating the numbers for both normal and autoplay modes!